### PR TITLE
fix up meta for proper 2.7 builds

### LINKED ISSUE #2: Unable to run help
Traceback of the issue shown below.

```
$ conda create -n cenv cookiecutter
Using Anaconda API: https://api.anaconda.org
Fetching package metadata ...............
Solving package specifications: ..........

Package plan for installation in environment /zopt/conda2/envs/cenv:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    chardet-2.3.0              |           py27_0         273 KB  defaults
    future-0.15.2              |           py27_0         639 KB  conda-forge
    poyo-0.3.0                 |           py27_0           9 KB  conda-forge
    whichcraft-0.4.0           |           py27_0           5 KB  conda-forge
    binaryornot-0.4.0          |           py27_0           7 KB  conda-forge
    arrow-0.7.0                |           py27_0          47 KB  conda-forge
    jinja2-time-0.1.0          |           py27_0           4 KB  conda-forge
    cookiecutter-1.4.0         |                0          39 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        1023 KB

The following NEW packages will be INSTALLED:

    arrow:           0.7.0-py27_0    conda-forge
    binaryornot:     0.4.0-py27_0    conda-forge
    ca-certificates: 2016.8.2-3      conda-forge
    certifi:         2016.8.2-py27_0 conda-forge
    chardet:         2.3.0-py27_0    defaults   
    click:           6.6-py27_0      conda-forge
    cookiecutter:    1.4.0-0         conda-forge
    future:          0.15.2-py27_0   conda-forge
    jinja2:          2.8-py27_1      conda-forge
    jinja2-time:     0.1.0-py27_0    conda-forge
    markupsafe:      0.23-py27_0     conda-forge
    ncurses:         5.9-8           conda-forge
    openssl:         1.0.2h-2        conda-forge
    poyo:            0.3.0-py27_0    conda-forge
    python:          2.7.12-0        conda-forge
    python-dateutil: 2.5.3-py27_0    conda-forge
    readline:        6.2-0           conda-forge
    setuptools:      25.1.6-py27_0   conda-forge
    six:             1.10.0-py27_0   conda-forge
    sqlite:          3.13.0-1        conda-forge
    tk:              8.5.19-0        conda-forge
    whichcraft:      0.4.0-py27_0    conda-forge
    zlib:            1.2.8-3         conda-forge

Proceed ([y]/n)? 

Fetching packages ...
chardet-2.3.0- 100% |#############################################################################################################| Time: 0:00:00   6.02 MB/s
future-0.15.2- 100% |#############################################################################################################| Time: 0:00:00  11.17 MB/s
poyo-0.3.0-py2 100% |#############################################################################################################| Time: 0:00:00   8.97 MB/s
whichcraft-0.4 100% |#############################################################################################################| Time: 0:00:00   4.39 MB/s
binaryornot-0. 100% |#############################################################################################################| Time: 0:00:00   6.38 MB/s
arrow-0.7.0-py 100% |#############################################################################################################| Time: 0:00:00   3.75 MB/s
jinja2-time-0. 100% |#############################################################################################################| Time: 0:00:00   4.16 MB/s
cookiecutter-1 100% |#############################################################################################################| Time: 0:00:00   3.25 MB/s
Extracting packages ...
[      COMPLETE      ]|################################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ]|################################################################################################################################| 100%
#
# To activate this environment, use:
# $ source activate cenv
#
# To deactivate this environment, use:
# $ source deactivate
#

$ source activate cenv
$ cookiecutter --help
Traceback (most recent call last):
  File "/zopt/conda2/envs/cenv/bin/cookiecutter", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/zopt/conda2/envs/cenv/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2953, in <module>
    @_call_aside
  File "/zopt/conda2/envs/cenv/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2939, in _call_aside
    f(*args, **kwargs)
  File "/zopt/conda2/envs/cenv/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2966, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/zopt/conda2/envs/cenv/lib/python2.7/site-packages/pkg_resources/__init__.py", line 635, in _build_master
    ws.require(__requires__)
  File "/zopt/conda2/envs/cenv/lib/python2.7/site-packages/pkg_resources/__init__.py", line 943, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/zopt/conda2/envs/cenv/lib/python2.7/site-packages/pkg_resources/__init__.py", line 829, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'cookiecutter==1.4.0' distribution was not found and is required by the application
```

### LINKED ISSUE #5: Using pip to install dependencies?
It appears we are using [pip](https://github.com/conda-forge/cookiecutter-feedstock/blob/b3e48df3f400d11ed126b8cf9b6029158b4bed07/recipe/run_test.sh#L1) to install dependencies that should be available (and do appear to be) from `conda-forge`. Is there a reason we cannot use those packages? What can we do so we are not using `pip` in the `test` step?

### PR DESCRIPTION
**Pull request by bollwyvl**
Fixes #2, hopefully!
Fixes #5

Not sure how this was even working before! This adds:
- `python` to all the build/run/test steps
- `setuptools` on build
- some command tests for `--version` and `--help`

Yeesh!

### CODE REVIEW

**Comment by willingc** [comment]
Nicely done @bollwyvl

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,27 +13,34 @@ source:
   sha256: 7697b1d64cd9b8eb5abd9e7594e318a80f1382f45d879c425944175697b29db2
 
 build:
-  number: 0
+  number: 1
   script: python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (jakirkham) [inline recipe/meta.yaml:17]: We might want to add [`entry_points`](http://conda.pydata.org/docs/building/meta-yaml.html#python-entry-points).

**Comment by jakirkham** [comment]
Might also be a good idea to re-render too. Have opened PR ( https://github.com/conda-forge/cookiecutter-feedstock/pull/4 ), which re-render's with the latest `conda-smithy`.

File: recipe/meta.yaml (hunk 1 above)
Comment (bollwyvl) [inline recipe/meta.yaml:17]: added!

**Comment by bollwyvl** [comment]
Added the entry_point...

Rerendered with smithy `1.0.3` on this PR so we don't have to increment the build a bunch of times...

Also removed the pip dependencies called out in #5, again for the same reason...

Here's hoping  🎰 !

**Comment by jakirkham** [comment]
Added a PR ( https://github.com/bollwyvl/cookiecutter-feedstock/pull/1 ) to your PR. Fixes up some other things. Some caused by doing the re-rendering. Others were related to the current testing strategy.

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipe/meta.yaml:17]: Thanks.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,27 +13,42 @@ source:
...
+  entry_points:
+    - cookiecutter = cookiecutter.cli:main
+    # master has changed it's setup a bit as of 4a64e48... will have to fix
+    # - cookiecutter = cookiecutter.__main__:main
```
Comment (jakirkham) [inline recipe/meta.yaml:21]: Ah, ok, that makes sense. Was wondering why I was having issues with it.

**Comment by bollwyvl** [comment]
@jakirkham thanks! Merged your simplification. That's great.

Also fixed the entrypoint to be the one associated with this release (not master) 😊

File: recipe/meta.yaml [hunk 3]
```diff
@@ -13,27 +13,42 @@ source:
...
+    - pytest-mock
+    - python
+    - freezegun
+  commands:
+    # Run basic commands.
+    - cookiecutter --version
+    - cookiecutter --help
+    # Run test suite.
+    - cd {{ environ["SRC_DIR"] }}
+    - py.test --cov=cookiecutter -k "not _hg_ and not mercurial and not test_make_sure_path_exists and not pypackage_git"
```
Comment (bollwyvl) [inline recipe/meta.yaml:51]: I guess we could actually test with mercurial on py27... not right now, though :]

**Comment by jakirkham** [comment]
Awesome! Thanks for your work on this.

Also, I'd be willing to help maintain if you wish to add me. Think I might be using this quite a bit in the near future.

File: recipe/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipe/meta.yaml:51]: Yeah, we could do some selector business. Seems fine to save it for a new issue and separate PR. This one is pretty busy anyways.

**Comment by bollwyvl** [comment]
@jakirkham of course, the more the merryer! I'll add you once the tests succeed/fail... hate giving up my spot in line at the appveyor ride 🎢 !

We're starting to fire up some more cookiecutter work over on Jupyter as well... granted, I do like the features afforded by the whole yeoman toolchain (mmm.... composability) but you can't beat cookiecutter for simplicity!

**Comment by jakirkham** [comment]
> @jakirkham of course, the more the merryer! I'll add you once the tests succeed/fail... hate giving up my spot in line at the appveyor ride 🎢 !

No worries. I can make a different PR for that.

Edit: Please see PR ( https://github.com/conda-forge/cookiecutter-feedstock/pull/6 ).

**Comment by bollwyvl** [comment]
@jakirkham Oh, it's probably worth adding you on this PR... tests have almost passed! hooray!

**Comment by jakirkham** [comment]
Ok, whichever is easiest. 😄

**Comment by jakirkham** [comment]
Passing. 😄 Not sure if you wanted to do more to it. Can go ahead and merge now if you want.

**Comment by bollwyvl** [comment]
Hooray test win!

I guess we'll pay the test tax one more time, so we can put this flurry of activity to bed 😓 !

**Comment by bollwyvl** [comment]
Hooray!

**Comment by jakirkham** [comment]
Marked the old builds as broken as it appears they didn't list the Python version. 😱 Fortunately, the new ones do thanks to these changes.

**Comment by bollwyvl** [comment]
Thanks!

**Pull request merged**